code/go/0chain.net/smartcontract/storagesc/shutdown.go

```diff
@@ -118,6 +118,26 @@ func (_ *StorageSmartContract) shutdownValidator(
 		return "", common.NewErrorf("shutdown_validator_failed", "can't get config: %v", err)
 	}
 
+	var refreshProviderFunc func(req provider.ProviderRequest) error
+	if actErr := cstate.WithActivation(balances, "hercules", func() error {
+		refreshProviderFunc = func(req provider.ProviderRequest) error {
+			stakePoolObject, err := getStakePool(spenum.Blobber, req.ID, balances)
+			if err != nil {
+				return err
+			}
+
+			stakePoolObject.TotalOffers = 0
+
+			return stakePoolObject.Save(spenum.Blobber, req.ID, balances)
+		}
+		return nil
+	}, func() error {
+		refreshProviderFunc = nil
+		return nil
```

Comment:
if you have an empty function here you could avoid nil check later on, it's a clearer way usually to havve a safe nil value

```go
	if p.IsKilled() || p.IsShutDown() {
		if refreshProvider != nil {
			err = refreshProvider(req)
			if err != nil {
				return err
			}
		}

		return AlreadyShutdownError
	}

```